### ISSUE TITLE
force:source:deploy fails with ENOENT -4058

### ISSUE THREAD

**Issue by Andrews-Consulting** [forcedotcom/cli#405]
I'm resubmitting this bug, as well as the work around I have implemented in my environment.

### Summary
When attempting to deploy certain aura components, the deployment will fail when the component is deployed via right click in VS Code (SFDX:Deploy source to Org) or via command line such as "sfdx force:source:deploy --sourcepath c:\...  force-app\main\default\aura\ApplyDiscounts\ApplyDiscounts.cmp --json --loglevel fatal". 
When issued in a terminal window, additional messages would display "[Error: ENOENT: no such file or directory, open 'C:\....AppData\Local\Temp\sdx_sourceDeploy_1589381165989.zip']
  errno: -4058,
  code: 'ENOENT',
  syscall: 'open',
  path:

### Additional information
sfdx-cli/7.57.1-dbf54fc584 win32-x64 node-v10.15.3
on Windows 10 Pro 1909 build 18363.778 with Ram drives

I have isolated the issue to  mdapiDeployApi.js.  in the function _zip, there is an assumption that when the archiver end event is received that the file storage system has closed the files and that the files are available to be read for deployment.   The archiver end event only signifies that the data is drained.  It appears that the directory entry doesn't exist until the FS close event has been fired.  I have a work around for my environment, since I'm always writing to a FS.  Of course, this is NOT a general purpose solution (if someone else looks at this work around).

```
const archive = archiver('zip', { zlib: { level: 9 } });
            archive.on('end', () => {
                this.logger.debug(`${archive.pointer()} bytes written to ${outFile} using ${this._getElapsedTime()}ms`);
                output.on('close', () => {resolve(outFile);});                  
                output.end();
            });
```

I haven't determined the cause of this FS behavior.  In general, it appears to be an issue with smaller files.

**Comment by Andrews-Consulting** [comment]
Had the same issue with 7.60.0-e2bb2907cb.  Carried my patch forward and it addressed the issue.

**Comment by Andrews-Consulting** [comment]
@mdonnalley This is still an issue and I have to carry my patch forward on each release.  Do you think this has been resolved by some other development effort? If so, which release carries the fix?

**Comment by ar-srinivas** [comment]
Is this issue fixed in the latest releases? We are using 7.179 and still facing this issue but intermittent.
